src/lazy/binary/raw/v1_1/value.rs

```diff
@@ -0,0 +1,251 @@
+    pub fn annotations(&self) -> RawBinaryAnnotationsIterator<'top> {
+        RawBinaryAnnotationsIterator::new(self.annotations_sequence())
+    }
+
+    /// Reads this value's data, returning it as a [`RawValueRef`]. If this value is a container,
+    /// calling this method will not read additional data; the `RawValueRef` will provide a
+    /// [`LazyRawBinarySequence_1_1`](crate::lazy::binary::raw::v1_1::sequence::LazyRawBinarySequence_1_1)
+    /// or [`LazyStruct`](crate::lazy::struct::LazyStruct) that can be traversed to access the container's contents.
+    pub fn read(&self) -> ValueParseResult<'top, BinaryEncoding_1_1> {
+        if self.is_null() {
+            let raw_value_ref = RawValueRef::Null(self.ion_type());
+            return Ok(raw_value_ref);
+        }
+
+        match self.ion_type() {
+            IonType::Null => unreachable!("all null types handled above"),
+            IonType::Bool => self.read_bool(),
+            IonType::Int => self.read_int(),
+            IonType::Float => self.read_float(),
+            IonType::Decimal => self.read_decimal(),
+            IonType::Timestamp => self.read_timestamp(),
+            IonType::Symbol => self.read_symbol(),
+            IonType::String => self.read_string(),
+            IonType::Clob => self.read_clob(),
+            IonType::Blob => self.read_blob(),
+            IonType::List => self.read_list(),
+            IonType::SExp => self.read_sexp(),
+            IonType::Struct => self.read_struct(),
+        }
+    }
+
+    /// Returns the encoded byte slice representing this value's data.
+    fn value_body(&self) -> IonResult<&'top [u8]> {
+        let value_total_length = self.encoded_value.total_length();
+        if self.input.len() < value_total_length {
+            eprintln!("[value_body] Incomplete {:?}", self);
+            return IonResult::incomplete(
+                "only part of the requested value is available in the buffer",
+                self.input.offset(),
+            );
+        }
+        let value_body_length = self.encoded_value.value_length();
+        let value_offset = value_total_length - value_body_length;
+        Ok(self.input.bytes_range(value_offset, value_body_length))
+    }
+
+    /// Returns an [`ImmutableBuffer`] containing whatever bytes of this value's body are currently
+    /// available. This method is used to construct lazy containers, which are not required to be
+    /// fully buffered before reading begins.
+    pub(crate) fn available_body(&self) -> ImmutableBuffer<'top> {
+        let value_total_length = self.encoded_value.total_length();
+        let value_body_length = self.encoded_value.value_length();
+        let value_offset = value_total_length - value_body_length;
+
+        let bytes_needed = std::cmp::min(self.input.len() - value_offset, value_body_length);
+        let buffer_slice = self.input.slice(value_offset, bytes_needed);
+        buffer_slice
+    }
```

Comment:
This is also part of that no-longer-true invariant. We should (as part of that `TODO`) find places where this is used and use `fn value_body` above instead.